### ISSUE TITLE
Fatal Error: Failed to initialize authentication interface.  Exiting...

### ISSUE THREAD

**Issue by formicsapien** [maxvollmer/Half-Life-VR#1]
**I don't really know how to build and run this,** but I downloaded a clone and built it with OpenVR libraries installed on my machine.  The build said it completed with no errors and 2 skipped.  After building I took the contents of the debug folder and pasted them in the "Valve" folder in Half-Life install directory.  When I just launch "HLVRLauncher.exe" the game runs with the console open and a bunch of stuff.  When I start a game there are some errors that pop up.  I assumed I needed to add the HLVRLauncher as a non-steam shortcut that runs in VR and I get this error:

"Fatal Error: Failed to initialize authentication interface.  Exiting..."

**EDIT**: I had it launch as administrator and it *will* boot the game, but it won't launch in VR.  Same thing happens as if I launched HLVRLauncher outside of steam.

**EDIT2:** I put the build contents into a "vr" folder in the Half-Life directory to make the game think it's a mod.  No changes.

Build output from visual studio:
1>------ Skipped Build: Project: hl, Configuration: Debug Win32 ------
1>Project not selected to build for this solution configuration 
2>------ Skipped Build: Project: cl_dll, Configuration: Debug Win32 ------
2>Project not selected to build for this solution configuration 
3>------ Build started: Project: opengl32, Configuration: Debug x64 ------
4>------ Build started: Project: HLVRLauncher, Configuration: Debug x64 ------
4>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\CL.exe /c /ZI /nologo /W3 /WX- /diagnostics:classic /Od /D _DEBUG /D _WINDOWS /D _UNICODE /D UNICODE /Gm /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Yc"stdafx.h" /Fp"x64\Debug\HLVRLauncher.pch" /Fo"x64\Debug\\" /Fd"x64\Debug\vc141.pdb" /Gd /TP /errorReport:prompt stdafx.cpp
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\CL.exe /c /ZI /nologo /W4 /WX- /diagnostics:classic /Od /D _DEBUG /D _WINDOWS /D _USRDLL /D GLPROXY_EXPORTS /D _WINDLL /D _UNICODE /D UNICODE /Gm /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"x64\Debug\\" /Fd"x64\Debug\vc141.pdb" /Gd /TP /errorReport:prompt opengl_proxy.cpp
4>stdafx.cpp
3>opengl_proxy.cpp
4>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\CL.exe /c /ZI /nologo /W3 /WX- /diagnostics:classic /Od /D _DEBUG /D _WINDOWS /D _UNICODE /D UNICODE /Gm /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Yu"stdafx.h" /Fp"x64\Debug\HLVRLauncher.pch" /Fo"x64\Debug\\" /Fd"x64\Debug\vc141.pdb" /Gd /TP /errorReport:prompt HLVRLauncher.cpp
4>HLVRLauncher.cpp
4>C:\Program Files (x86)\Windows Kits\10\bin\10.0.15063.0\x86\rc.exe /D _UNICODE /D UNICODE /l"0x0409" /nologo /fo"x64\Debug\HLVRLauncher.res" HLVRLauncher.rc
3>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\link.exe /ERRORREPORT:PROMPT /OUT:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.dll" /INCREMENTAL /NOLOGO kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib /DEF:"opengl_proxy.def" /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG:FASTLINK /PDB:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.lib" /MACHINE:X64 /DLL x64\Debug\opengl_proxy.obj
4>C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.10.25017\bin\HostX86\x64\link.exe /ERRORREPORT:PROMPT /OUT:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\HLVRLauncher.exe" /INCREMENTAL /NOLOGO kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib /MANIFEST /MANIFESTUAC:"level='asInvoker' uiAccess='false'" /manifest:embed /DEBUG:FASTLINK /PDB:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\HLVRLauncher.pdb" /SUBSYSTEM:WINDOWS /TLBID:1 /DYNAMICBASE /NXCOMPAT /IMPLIB:"C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\HLVRLauncher.lib" /MACHINE:X64 x64\Debug\HLVRLauncher.res
4>x64\Debug\HLVRLauncher.obj
4>x64\Debug\stdafx.obj
4>HLVRLauncher.vcxproj -> C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\HLVRLauncher.exe
4>HLVRLauncher.vcxproj -> C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\HLVRLauncher.pdb (Partial PDB)
3>   Creating library C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.lib and object C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.exp
3>GLProxy.vcxproj -> C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.dll
3>GLProxy.vcxproj -> C:\Users\formi\Desktop\Half-Life-VR-master\x64\Debug\opengl32.pdb (Partial PDB)
**========== Build: 2 succeeded, 0 failed, 0 up-to-date, 2 skipped ==========**

**Comment by wiiboy123** [comment]
I get the same error when I run HLVR without the launcher and the DLLs in place, except this error appears on startup. When I run the launcher, literally nothing happens (not even a Task Manager entry).

**Comment by maxvollmer** [comment]
You need to build the client and server dlls. No idea why they are skipping, but you definitely need to build them. You need to create a mod folder in your Half-Life directory with a liblist.gam (see any HL1 mod tutorial out there, or just take a liblist.gam from another mod and modify it). client.dll and hl.dll go into their respective folders inside your mod folder. HLVRLauncher.exe and opengl32.dll go directly into your mod folder. Do not put the opengl32.dll into your Half-Life folder. Do not run the game without the launcher (the launcher puts opengl32.dll into the Half-Life folder before starting the game and makes sure that opengl32.dll is deleted after the game ends). Do not play online. Doing this wrong might get you VAC banned.

**Comment by wiiboy123** [comment]
The game now starts with the HLVRLauncher EXE, but the game is not in VR. I've tried starting it with SteamVR on and without it on. I built the DLLs and made the mod folder. I should note that when starting a game, it says "Unknown command: VModEnable"

**Comment by formicsapien** [comment]
Yeah, I'm not building it right in Visual Studio.  I've only ever built like 3 projects, so I'm probably doing something wrong.  I'm hitting Build>Build Solution  and I think it's only building the launcher component.

**Comment by wiiboy123** [comment]
@formicsapien Right click each project individually and click "Rebuild"

**Comment by formicsapien** [comment]
@wiiboy123 Yeah, I did that.  but it still dumped all the build files into individual "Debug" folders.  I have no idea if that's normal, but I copied and pasted all of it into the mod folder.  Now, I'm having another issue that I've opened.

**Comment by wiiboy123** [comment]
@formicsapien Yeah, that's normal. It's Visual Studio just dumping the output files into different directories since they're all separate projects. Just collect all the files and put them in their respective folders in your mod folder.

**Comment by formicsapien** [comment]
@wiiboy123 I can't click on the disclaimer window buttons.  It asks if I'm accept the dev has no liability, did you get that far?

**Comment by wiiboy123** [comment]
@formicsapien No, I never saw that message.
